### ISSUE TITLE
`cargo-sort` wants to remove lines before comments

### ISSUE THREAD

**Issue by smoelius** [DevinR528/cargo-sort#57]
As of recent, `cargo-sort` wants to remove lines before comments.

For example, `cargo-sort` requires that the following lines be removed to consider their enclosing Cargo.toml file sorted:
* https://github.com/trailofbits/test-fuzz/blob/67c8004af9cc6a295143dc537eb725cf44d06cc0/runtime/Cargo.toml#L11
* https://github.com/trailofbits/test-fuzz/blob/67c8004af9cc6a295143dc537eb725cf44d06cc0/test-fuzz/Cargo.toml#L29

**Comment by DevinR528** [comment]
Does this also fail when the `--grouped` option is passed? This is, in general, a hard thing to do without either needing several settings or making decisions for the user.

```
# does this comment belong above or below
key = value # to the left or below
# above or below
```
Same with tables.

**Comment by smoelius** [comment]
> Does this also fail when the `--grouped` option is passed?

Yes. [Here](https://github.com/trailofbits/test-fuzz/blob/68330e356a3eb41339eb489e773cd42bb891b558/.github/workflows/ci.yml#L56) specifically is how I run it:
```sh
find . -name Cargo.toml | xargs -n 1 dirname | xargs -n 1 cargo sort --check --grouped
```

**Comment by smoelius** [comment]
>This is, in general, a hard thing to do without either needing several settings or making decisions for the user.

I can appreciate that. But perhaps the `--check` case could be treated specially?

**Comment by smoelius** [comment]
I think that could work.

Mainly, I just want a command like [I posted](https://github.com/DevinR528/cargo-sort/issues/57#issuecomment-1316102874) to succeed, i.e., I am really interested in the `--check` option.

> While adding options is not my favorite idea, I really don't see away around this without changing the current behavior for all users.

To be clear, the command I posted succeeded until recently. So, I think something did change.

Thank you very much for looking into this. I value `cargo-sort` and use it in CI for several projects.

**Comment by DevinR528** [comment]
> To be clear, the command I posted succeeded until recently. So, I think something did change.

Ahh, this is helpful. In the next week, I will spend some time figuring out when this broke, and hopefully, that will lead to a fix.
(I have a break coming up at the end of this week, so I should have time to work on this)

**Comment by Alchiadus** [comment]
`cargo init` will insert a comment by default, which also triggers this issue.

A workaround is to use `cargo sort --check --no-format`, rather than just `cargo sort --check`:

```
$ cargo init foobar
     Created binary (application) package

$ cd foobar

$ cargo sort --check
Checking foobar...
error: Cargo.toml for foobar is not formatted

$ cargo sort --check --no-format
Checking foobar...
```

`Cargo.toml`:
```toml
[package]
name = "foobar"
version = "0.1.0"
edition = "2021"

# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html

[dependencies]
```

**Comment by smoelius** [comment]
Thanks, @Alchiadus! I don't know how I overlooked that.
